client/src/App.js

```diff
@@ -26,85 +28,80 @@ class App extends Component {
       <div>
         <Navbar fluid>
           <Navbar.Header>
-            <Navbar.Brand>
-              <a href="#">BitsPlease</a>
+            <Navbar.Brand href='home' className='logo-text'>
+              BitsPlease
             </Navbar.Brand>
-            {
-              isAuthenticated() && (
-                <null>
+            {isAuthenticated() && (
+              <null>
                 <Button
-                bsStyle="primary"
-                className="btn-margin"
-                onClick={this.goTo.bind(this, 'home')}
-              >
-                Home
-              </Button>
-                
-                  <Button
-                    bsStyle="primary"
-                    className="btn-margin"
-                    onClick={this.goTo.bind(this, 'profile')}
-                  >
-                    Profile
-                  </Button>
-                  </null>
-                )
-            }
-            {
-              isAuthenticated() && (
+                  bsStyle='primary'
+                  className='btn-margin'
+                  onClick={this.goTo.bind(this, 'home')}
+                >
+                  Home
+                </Button>
+
                 <Button
-                bsStyle="primary"
-                className="btn-margin"
-                onClick={this.goTo.bind(this, 'some-bits-Please')}
+                  bsStyle='primary'
+                  className='btn-margin'
+                  onClick={this.goTo.bind(this, 'profile')}
+                >
+                  Profile
+                </Button>
+              </null>
+            )}
+            {isAuthenticated() && (
+              <Button
+                id='qsLogoutBtn'
+                bsStyle='primary'
+                className='btn-margin'
+                onClick={this.logout.bind(this)}
               >
-                some bits-Please!
+                Log Out
               </Button>
-                )
-            }
-            {
-              isAuthenticated() && (
-                  <Button
-                    id="qsLogoutBtn"
-                    bsStyle="primary"
-                    className="btn-margin"
-                    onClick={this.logout.bind(this)}
-                  >
-                    Log Out
-                  </Button>
-                )
-            }
+            )}
+            {isAuthenticated() && (
+            <Form class="form-inline">
+              <FormControl class="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search"></FormControl>
+              <Button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</Button>
+            </Form>
+            )}
+            {isAuthenticated() && (
+              <Avatar
+                googleId={Auth.idToken}
```

Comment:
I'm definitely referencing this wrong, but the general idea would be to put the idToken here and I wasn't sure how you were passing it around, so I took a stab at it, but feel free to correct me and tell me how it's done. :)